packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
Yeah, I understand that violation, and alluded to it in my comment above.

However, my main point is that I think the current signature is not very useful.  Not by my standards (`val + val => val`) and not even by @SeanTAllen's standards, since it isn't compatible with the signature of `String.add`:

```pony

actor Main
  new create(env': Env) =>
    let a: ByteSeq val = recover val [as U8: 1, 2, 3] end
    let b: ByteSeq val = "foo"
    a + b
```
```pony
Error:
/home/jemc/1/code/gitx/ponyc/test/jemc/test.pony:73:7: a member of the union type has an incompatible method signature
    a + b
      ^
Error:
/home/jemc/1/code/gitx/ponyc/packages/builtin/string.pony:940:3: first implementation is here
  fun add(that: String box): String =>
  ^
Error:
/home/jemc/1/code/gitx/ponyc/packages/builtin/array.pony:279:3: second implementation is here
  fun add(that: this->Array[A]): Array[this->A!] =>
```